djangocms_navigation/admin.py

```diff
@@ -1,26 +1,52 @@
 from django.conf.urls import url
 from django.contrib import admin
+from django.contrib.admin.views.main import ChangeList
 from django.contrib.sites.shortcuts import get_current_site
 from django.shortcuts import reverse, HttpResponseRedirect
+from django.contrib.admin.utils import quote
+
 from django.utils.text import slugify
 from django.utils.html import format_html
 from django.utils.translation import ugettext_lazy as _
 
 from treebeard.admin import TreeAdmin
-from treebeard.forms import movenodeform_factory
+
 from .models import Menu, MenuContent, MenuItem
+from .forms import MenuItemForm, MenuContentForm
+
+
+class MenuItemChangeList(ChangeList):
+    def __init__(self, request, *args, **kwargs):
+        self.menu_content_id = request.menu_content_id
+        super().__init__(request, *args, **kwargs)
+
+    def url_for_result(self, result):
+        pk = getattr(result, self.pk_attname)
+        return reverse(
+            "admin:%s_%s_change" % (self.opts.app_label, self.opts.model_name),
+            args=(self.menu_content_id, quote(pk)),
+            current_app=self.model_admin.admin_site.name,
+        )
+
+    # def get_queryset(self, request):
+    #     self.request = request
+    #     return super().get_queryset(request)
 
 
 class MenuContentAdmin(admin.ModelAdmin):
-    exclude = ["menu"]
+    form = MenuContentForm
     list_display = ["title", "get_menuitem_link"]
+    list_display_links = None
 
     def save_model(self, request, obj, form, change):
         if not change:
+            title = form.data.get("title")
```

Comment:
I think I missed this before... This is actually potentially a security risk - `form.data` is the data as it came in the POST. What you want to be getting is `form.cleaned_data`.